components/scream/src/share/tests/vertical_interp_tests.cpp

```diff
@@ -11,46 +11,192 @@ using Spack = SmallPack<Real>;
+  }//end of looping over perturbed or not
+}
 
-  auto npacks_src = ekat::PackInfo<1>::num_packs(n_layers_src);
-  auto npacks_tgt = ekat::PackInfo<1>::num_packs(n_layers_tgt);
-  auto p_tgt = view_1d<Pack<Real,1>>("",npacks_tgt);
+TEST_CASE("main_vertical_interpolation_test"){
+  //This test first checks that you get the same answer if source and target levels 
+  //are the same
+  //It then tests that you get the expected answer for nlevs % packsize==0 and nlevs%packsize!=0
+  //and for different pack sizes. 
+  run<1> ();
+  if (SCREAM_PACK_SIZE>1) {
+    run<SCREAM_PACK_SIZE> ();
+  }
+}
+
+TEST_CASE("testing_masking"){
+  //This test performs 3 tests:
+  //1) That the interpolation is working properly using 2d views, 
+  //   including the masking of out-of-bounds values
+  //2) It checks the interpolation is working properly with 
+  //   a user defined masking value
+  //3) It checks that the interpolation is working properly when 
+  //   using the 1d interpolation function
+  const int n_layers_src = 9;
+  const int n_layers_tgt = 17;
+  const int N = SCREAM_PACK_SIZE;
+
+  auto npacks_src = ekat::PackInfo<N>::num_packs(n_layers_src);
+  auto npacks_tgt = ekat::PackInfo<N>::num_packs(n_layers_tgt);
+  auto p_tgt = view_1d<Pack<Real,N>>("",npacks_tgt);
   auto p_tgt_s = Kokkos::create_mirror_view(ekat::scalarize(p_tgt));
-  auto tmp_src = view_2d<Pack<Real,1>>("",3,npacks_src);
+  auto tmp_src = view_2d<Pack<Real,N>>("",2,npacks_src);
   auto tmp_src_s = Kokkos::create_mirror_view(ekat::scalarize(tmp_src));
-  auto p_src = view_2d<Pack<Real,1>>("",3,npacks_src);
+  auto p_src = view_2d<Pack<Real,N>>("",2,npacks_src);
   auto p_src_s = Kokkos::create_mirror_view(ekat::scalarize(p_src));
-  auto out = view_2d<Pack<Real,1>>("",3,npacks_tgt);
+  auto out = view_2d<Pack<Real,N>>("",2,npacks_tgt);
   auto out_s = Kokkos::create_mirror_view(ekat::scalarize(out));
-  auto mask = view_2d<Mask<1>>("",3,npacks_tgt);
+  auto mask = view_2d<Mask<N>>("",2,npacks_tgt);
 
-  //Test to see if interpolate properly using 2d views
+  //Fist test to see if interpolate properly using 2d views
   //Also test that when out-of-bounds returns masked values
-  p_tgt_s(0) = 20.0;
-  p_tgt_s(1) = 25.0;
-  p_tgt_s(2) = 35.0;
-  p_tgt_s(3) = 45.0;
-
-  p_src_s(0,0) = 20.0;
-  p_src_s(0,1) = 30.0;
-  p_src_s(0,2) = 40.0;
-
-  p_src_s(1,0) = 30.0;
-  p_src_s(1,1) = 40.0;
-  p_src_s(1,2) = 50.0;
 
-  p_src_s(2,0) = 10.0;
-  p_src_s(2,1) = 20.0;
-  p_src_s(2,2) = 30.0;
-
-  for (int i=0; i<3; i++){
-    tmp_src_s(i,0)=260.;
-    tmp_src_s(i,1)=270.;
-    tmp_src_s(i,2)=240.;
+  //Set target levels from 25-105
+  for (int i=0; i<17; i++){
+    p_tgt_s(i) = 25.0 + i*5.0;
+  }
+ 
+  //Set source levels:
+  //For column 1 from 20-100
+  //For column 2 from 30-110
+  //Set input variable, in this case
+  //set tmp_src from 200-280 by 10, with final level at 240 so 
+  //can test interpolation with increasing/decreasing values
+  for (int i=0; i<9; i++){
```

Comment:
After this loop, you have to deep copy the inputs to the device views.

Reply:
Added deep copy